### PR TITLE
Add default config location

### PR DESCRIPTION
**Pull request by Leon-Plickat**
As discussed in #147.

With this patch, if a config path is not provided by user, river tries `$XDG_CONFIG_HOME/river/init` and `/etc/river/init`.

#### TODO
* [x] Find out how `std.os.access()` works (definitely not like the C version).
* [x] Find out why zig complains about freeing `startup_command`.

### CODE REVIEW

File: river/main.zig [hunk 1]
```diff
@@ -32,7 +32,21 @@ const usage: []const u8 =
     \\
 ;
 
+fn getStartupCommand(allocator: *std.mem.Allocator) anyerror!?[*:0]const u8 {
+    if (std.os.getenv("XDG_CONFIG_HOME")) |xdg_config_home| {
+        const path = try std.fmt.allocPrintZ(allocator, "{}/river/init", .{xdg_config_home});
+        if (try std.os.access(path, 1)) return path; // 1 == X_OK
```
Comment (ifreund) [inline river/main.zig:38]:
```suggestion
        if (try std.os.access(path, std.os.X_OK)) return path;
```

**Review by ifreund — COMMENTED** [review]
For using `os.access()` you probably want to get rid of the ifs and just do 
```zig
std.os.access(foo, std.os.X_OK) catch return null;
```

**Comment by Leon-Plickat** [comment]
Almost there.

Only thing left to do is the type. If I use a terminated string (`[*:0]const u8` or `[:0]const u8`) for the path (`startup_command`), then zig complains about freeing it, but if I use regular `[]const u8`, then it can't be integrated into `child_args` because it is not terminated.

File: river/main.zig [hunk 2]
```diff
@@ -32,8 +32,42 @@ const usage: []const u8 =
...
+    if (try testConfigPath(allocator, "/etc/river/init", .{})) |path| {
+        return path;
+    }
+    return null;
+}
+
 pub fn main() anyerror!void {
-    var startup_command: ?[*:0]const u8 = null;
+    var allocator_instance = std.heap.GeneralPurposeAllocator(.{}){};
+    const gpa = &allocator_instance.allocator;
```
Comment (ifreund) [inline river/main.zig:69]: really we should just use `util.gpa` here like the rest of the code base.

**Review by ifreund — COMMENTED** [review]
I think `[:0]const u8` is the type you want here, though I don't see why zig would complain about freeing it. What's the error that you get?

**Comment by Leon-Plickat** [comment]
I figured it out: Zig did not knew that `startup_command` could not be `null` at that point, so I had to explicitly tell it that via `free(startup_command orelse unreachable)`.

**Comment by Leon-Plickat** [comment]
Ready for review.

File: river/main.zig [hunk 3]
```diff
@@ -32,8 +32,36 @@ const usage: []const u8 =
...
+            return path;
+        }
+    } else if (std.os.getenv("HOME")) |home| {
+        if (try testConfigPath("{}/.config/river/init", .{home})) |path| {
+            return path;
+        }
+    }
+    if (try testConfigPath("/usr/local/etc/river/init", .{})) |path| {
+        return path;
+    }
```
Comment (ifreund) [inline river/main.zig:56]: I don't see a reason we should look here. Do you have a strong reason why we should do so?
```suggestion
```

File: river/main.zig [hunk 4]
```diff
@@ -82,6 +124,7 @@ pub fn main() anyerror!void {
             if (std.os.system.sigprocmask(std.os.SIG_SETMASK, &std.os.empty_sigset, null) < 0) unreachable;
             std.os.execveZ("/bin/sh", &child_args, std.c.environ) catch c._exit(1);
         }
+        util.gpa.free(startup_command orelse unreachable);
```
Comment (ifreund) [inline river/main.zig:127]:
```suggestion
        util.gpa.free(cmd);
```

File: river/main.zig (hunk 3 above)
Comment (Leon-Plickat) [reply river/main.zig:56]: Convention. "Manually" installed things go to `/usr/local/`, which becomes relevant if we decide to ship a default config (per #146, f.e.).

Comment (ifreund) [reply river/main.zig:56]: I think manually installing stuff system wide is an antipattern and people should use `-/.local/` for that use case. (I should probably amend the build command example in the readme I realize).

Comment (ifreund) [reply river/main.zig:56]: The exact system dir is something we may want to make configurable at build time in the future though, I'm sure there's at least one distro out there that doesn't use `/etc/`

Comment (Leon-Plickat) [reply river/main.zig:56]: Sounds reasonable. I'll remove the `/usr/local/` part for now.

**Review by ifreund — APPROVED** [review]
Thanks!

**Pull request merged**